src/test/osd/TestOSDMap.cc

```diff
@@ -979,7 +979,7 @@ TEST_F(OSDMapTest, CleanPGUpmaps) {
         int new_up_primary;
         osdmap.pg_to_raw_up(pgid, &new_up, &new_up_primary);
         ASSERT_TRUE(new_up == up);
-        ASSERT_TRUE(new_up_primary = up_primary);
+        ASSERT_EQ(new_up_primary, up_primary);
```

Comment:
again, i'd suggest use 
```c++
ASSERT_EQ(up_primary, new_up_primary);
```

see the definition of `ASSERT_EQ`
```c++
#define GTEST_ASSERT_EQ(expected, actual) \
  ASSERT_PRED_FORMAT2(::testing::internal:: \
                      EqHelper<GTEST_IS_NULL_LITERAL_(expected)>::Compare, \
                      expected, actual)
```
the first parameter is the expected value. in this case, it's `up_primary`.

also, could you take this opportunity to update the line above?

Reply:
sorry, i misundestood 'the line above'